Sources/Shared/TCPSocket.cpp

```diff
@@ -42,7 +42,9 @@ int TCPSocket::send(const std::string &data) {
   return ::send(this->socketDescriptor, data.c_str(), data.size(), 0);
 }
 
-int TCPSocket::receive(std::string &data) {
+int TCPSocket::receive(std::string &data, int bytes) {
+  char buffer[bytes];
```

Comment:
Shouldn't this be dynamically allocated? char* buffer = new char[bytes];
And then deallocated at the end: delete []buffer;